client/lib/gsuite/get-google-expiry-date.ts

```diff
@@ -0,0 +1,5 @@
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+export function getGoogleExpiryDate( domain: ResponseDomain ): string | null {
+	return domain.googleAppsSubscription?.expiryDate ?? null;
```

Comment:
It may also be worth allowing `domain` to be undefined in the logic as well.
```suggestion
	return domain?.googleAppsSubscription?.expiryDate ?? null;
```